quickwit-doc-mapper/src/doc_mapper.rs

```diff
@@ -43,6 +44,21 @@ pub trait DocMapper: Send + Sync + Debug + DynClone + 'static {
     /// (we pass by value here, as the value can be used as is in the _source field.)
     fn doc_from_json(&self, doc_json: String) -> Result<Document, DocParsingError>;
 
+    /// Converts a tantivy named Document to the json format.
+    ///
+    /// Tantivy does not have any notion of cardinality nor object.
+    /// It is therefore up to the `DocMapper` to pick a tantivy named document
+    /// and convert it into a final quickwit document.
+    ///
+    /// Because this operation is dependent on the `DocMapper`, this
+    /// method is meant to be called on the root node using the most recent
+    /// `DocMapper`. This ensures that the different hits are formatted according
+    /// to the same schema.
+    fn doc_to_json(
```

Comment:
We could consider custom payloads in the tantivy docstore. This could be done by converting `Document` to a trait, that e.g. implements doc_to_json or similar

Reply:
I added something here https://github.com/quickwit-oss/tantivy/issues/1352

Reply:
thank you!